src/main/java/org/javarosa/core/model/instance/DataInstance.java

```diff
@@ -279,7 +279,7 @@ public void setID(int recordid) {
         this.recordid = recordid;
     }
 
-    public abstract DataInstance initialize(InstanceInitializationFactory initializer, String instanceId);
+    public abstract DataInstance initialize(InstanceInitializationFactory initializer, String instanceId, String locale);

```

Comment:
Could you not create a concrete method with the original signature which calls the abstract one with the locale being null? Then you would not have to pass the null everywhere.

Reply:
Very true! This did indeed get rid of some unnecessary null usage 🎉